src/app/docs/id/id-docs.component.html

```diff
@@ -100,8 +90,10 @@
       >
         <p>
           Don't use the ID directive when an element does not need to be referenced by its
-          ID attribute, since doing so adds unnecessary overhead to an application. The same rule
-          goes for assigning a hard-coded ID to an element, because it can lead to ID collisions.
+          ID attribute.
+        </p>
+        <p>
+          Don't use the ID directive to hard-code an ID to element, which can lead to ID collisions.
```

Comment:
The ID directive can't be used to hard-code an ID. The unmodified text is saying that in addition to not using the ID directive unnecessarily, you should also not use the conventional method of assigning an ID (by setting the element's `id` attribute

Reply:
How about `Don't use the ID directive in combination with hard-coded IDs because this can cause collisions. We recommend only using the ID directive.`

Reply:
That's still not the original intent. What I'm trying to convey is that you shouldn't use the ID directive when the generated ID doesn't need to be referenced by another element, _and_ you should never hard-code an ID, either. 

Example of using the ID directive unnecessarily (bad):

```
<!-- I have no label element or any other element that refers to this element, but I'm still using skyId to generate an element. I don't need to do this. -->
<input skyId>
```

Example of hard-coding an ID (bad):

```
<!-- I'm using a hard-coded ID so the label can reference the input. This is bad because there might be another element with an ID of "some-id" which will lead to an ID collision. Instead I should use the ID directive. -->
<label for="my-id">My label</label>
<input id="my-id">
```

Example of using the ID directive to associate two elements (good):

```
<label [for]="myInput">My label</label>
<input skyId #myInput"skyId">
```

Reply:
OK. I was trying to keep the guidance wording centered on the ID directive for consistency with guidelines elsewhere, but we can reframe things. How about:
 
```Don't use hard-coded IDs because they can cause collisions with the IDs that the ID directive creates. Only use the ID directive.```

Reply:
That's still different from the original intent. It's unlikely that a consumer will specify an ID that would clash with an ID generated by the directive; the problem we're trying to avoid is when two hard-coded IDs collide. Developers don't always have control over the entire page, so they can't guarantee a hard-coded ID will ever be unique. The ID directive _can_ guarantee that the ID will be unique.

Reply:
How about:
```Don't assign hard-coded IDs to elements because this causes collisions if IDs are not unique. Always use the ID directive to assign IDs because it guarantees that IDs are unique.```

Reply:
That last version is perfect. Thanks.